src/app/route/MyAccountDetails/MyAccountDetails.component.js

```diff
@@ -0,0 +1,630 @@
+                            >
+                                Edit Address
+                            </button>
+                        </div>
+                    </>
+                );
+            }
+        }
+
+        return (
+            <>
+                <div>
+                    No default&nbsp;
+                    { addressType }
+                    &nbsp;address has been set.
+                </div>
+                <div block="MyAccountDetails" elem="Actions">
+                    <button
+                      block="Button"
+                      mods={ { likeLink: true } }
+                      onClick={ () => this.changeState(STATE_UPDATE_ADDRESS, addressType) }
+                    >
+                        Add New Address
+                    </button>
+                </div>
+            </>
+        );
+    }
+
+    /**
+     * Render Custoenr Address Book block
+     */
+    renderAddressBook() {
+        return (
+            <fieldset block="MyAccountDetails" elem="AddressBook">
+                <legend>
+                    <span>Address Book</span>
+                </legend>
+                <div block="MyAccountDetails" elem="FieldWrapper">
+                    <div block="MyAccountDetails" elem="AddressWrapper">
+                        <div block="MyAccountDetails" elem="FieldWrapper">
+                            <h4>Default Billing Address</h4>
+                            { this.renderAddress('billing') }
+                        </div>
+                        <div block="MyAccountDetails" elem="FieldWrapper">
+                            <h4>Default Shipping Address</h4>
+                            { this.renderAddress('shipping') }
+                        </div>
+                    </div>
+                </div>
+            </fieldset>
+        );
+    }
+
+    render() {
+        const { state } = this.state;
+        const renderFunction = this.renderMap[state];
+        const { customer } = this.props;
+
+        if (customer && Object.keys(customer).length) {
+            return (
+                <main block="MyAccountDetails" aria-label="My Account Details">
+                    <div block="MyAccountDetails" elem="Wrapper">
+                        <ul block="MyAccountDetails" elem="Sidebar">
+                            <li
+                              onClick={ () => this.changeState(STATE_ACCOUNT_OVERVIEW) }
+                              onKeyPress={ () => this.changeState(STATE_ACCOUNT_OVERVIEW)}
+                            >
+                                My Account
+                            </li>
+                            <li>My Orders</li>
+                        </ul>
+                        <div block="MyAccountDetails" elem="Content">
+                            { renderFunction() }
+                        </div>
+                    </div>
+                </main>
+            );
+        }
+
+        return (
```

Comment:
I do not like this. Why can not we use the same functions, just were not you are about to render data use `TextPlaceholder`?